### PR TITLE
Features/update asset roles

### PR DESCRIPTION
**Pull request by anilnatha**
## Purpose

Updated dataset asset lists so that we now use the URI/HREF of the asset as the key in the list the assets. The role/type of the data (e.g. "metadata", "data", etc.) is now an attribute of the asset with a key of "roles".

## Proposed Changes
- [CHANGE] Use href as unique keys for list of assets instead of asset role
## Issues
- #67 
- #69 
## Testing
- Tested using built-in unit tests

### CODE REVIEW

File: unity_sds_client/services/data_service.py [hunk 1]
```diff
@@ -66,8 +66,14 @@ def get_collection_data(self, collection: type= Collection):
...
         for dataset in results:
             ds = Dataset(dataset['id'], collection.collection_id, dataset['properties']['start_datetime'], dataset['properties']['end_datetime'], dataset['properties']['created'])
-            ds.add_data_file(DataFile("data" ,dataset['assets']['data']['href']))
-            ds.add_data_file(DataFile("metadata" ,dataset['assets']['metadata__data']['href']))
+
+            for filename in dataset['assets']:
+                location = dataset['assets'][filename]['href']
+                file_type = dataset['assets'][filename]['type']
+                title = dataset['assets'][filename]['title']
+                description = dataset['assets'][filename]['description']
```
Comment (mike-gangl) [inline unity_sds_client/services/data_service.py:74]: Type, Title, and Description are all optional elements. What will happen here if they aren't provided?

Comment (mike-gangl) [reply unity_sds_client/services/data_service.py:74]: The problem with this code block is we are parsing STAC from U-DS differently than we are parsing STAC from the to/from_stac methods. in the to/from_stac methods, we map the DataFile.type property to an asset_role, here we are mapping it to the asset.type property, which are not the same and will cause issues here. (e.g. the 'type' of a file might be "application/netcdf" but if we call  "to_stac" on that, we'll get some bizarre asset-roles.

here we should either treat this like the CMR query (where we look for asset roles and if they are there use them, if not, check the "key" if it looks like "data" or "metadata" then make that the filetype, but we cannot mix/match these.

**Comment by anilnatha** [comment]
Hey @mike-gangl I've made the necessary updates, and tested using both the unit tests (all passed) and the `5_working_with_chirp_data.ipynb` tutorial notebook. please review at your earliest.

I was being cautious with trying not to conflate the role and type information. The changes I've made work for CMR presently, and my hope is that they will also work if/when changes are made to CMR to adopt the usage of role information in the same manner that we will be using it with stac.

The information you provided through this effort really helped me better understand our usage of Collections, datasets, and datafiles. 👍🏽

**Comment by anilnatha** [comment]
Oh and the checks on Github are failing because calls to SPS are returning a status of `500`. will need to look into why that is happening.

**Review by mike-gangl — APPROVED** [review]

**Pull request merged**